ui/imports/shared/views/chat/LinksMessageView.qml

```diff
@@ -210,7 +214,7 @@ Column {
             Component.onCompleted: {
                 // putting this is onCompleted prevents automatic binding, where
                 // QML warns of a binding loop detected
-                this.sourceComponent = getSourceComponent()
+                Qt.callLater((instance) => { instance.sourceComponent = getSourceComponent() }, this)
```

Comment:
Another improvement we could do later is do the `getLinkDataForStatusLinks` and other stuff on the backend, so we only send the final JSON stringified in the model.

Reply:
Yes, the LinksMessageView deserves some bigger cleanup too (not necessarily in the PR though)

Reply:
Yeah let's do that in another PR. This one is already massive :smile:

Reply:
I'll drop these changes and create another task for LinksMessageView. Just found that sometimes when scrolling fast enough the callback in Qt.callLater is called in an invalid context. Probably the delegate is already destroyed.

Reply:
Added another task for it: #9101